ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -72,6 +72,24 @@ static inline void bc_fast_mul(bc_num n1, size_t n1len, bc_num n2, size_t n2len,
 	}
 }
 
+static inline void bc_fast_square(bc_num n1, size_t n1len, bc_num *prod)
```

Comment:
Please add a comment above the function that this is for the case where the number is so short that you can do it without an array.